homeassistant/helpers/entity.py

```diff
@@ -252,10 +252,8 @@ def _async_write_ha_state(self):
             else:
                 state = str(state)
 
-            attr = self.state_attributes or {}
-            device_attr = self.device_state_attributes
-            if device_attr is not None:
-                attr.update(device_attr)
+            attr.update(self.state_attributes or {})
```

Comment:
`self.state_attribute` is a property and should always return a new object, eg a new dict.

With the new code we're avoiding a potential bug if something would not follow our architecture. It's not a bug today, though.